### ISSUE TITLE
Map tips changed behaviour

### ISSUE THREAD

**Issue by northwestview** [qgis/QGIS#31311]
In version 3.8.1, the map tips dialog stays open when the pointer is moved off of the point, allowing any links in the dialog to be clicked on. In 3.8.2 as soon as the pointer is moved off of the point with the map tip, the map tip dialog disappears. If there is a setting to set the length of time the map tip stays open, it should be noted somewhere in the documentation.

**Comment by gioman** [comment]
>  If there is a setting to set the length of time the map tip stays open, it should be noted somewhere in the documentation

there is: configuration > options > canvas and legend > map tips

https://docs.qgis.org/testing/en/docs/user_manual/introduction/qgis_configuration.html?highlight=map%20tips#canvas-and-legend-settings

**Comment by northwestview** [comment]
Thank you for the response.

That setting only seems to affect how long it takes for the map tip to
open, not how long it stays open. As I mentioned, in 3.8.2 as soon as I
move the pointer off of the point that has the map tip, the map tip
disappears. In 3.8.1 the map tip stayed open when moving the pointer off
the point with the map tip. Having the map tip disappear means that any
clickable links in the map tip cannot be clicked on (to open an image for
instance).

There is a work-around: by right-clicking on the point after the map tip
opens. This opens a dialog for the selection of which layer to show the
identify feature dialog for, then it is possible to left click in the map
tip dialog and the map tip dialog stays open so that links can be clicked.
Much different than the way the map tip functioned in 3.8.1.

**Comment by gioman** [comment]
I confirm it, is no longer possible to have a maptip stay on open when moving a little the mouse, so is not possible anymore to click on (i.e.) links.

**Comment by gioman** [comment]
The change appeared in 3.8.2 (works ok on 3.8.1) and affects also 3.4.11 (works ok on 3.4.10).

**Comment by nirvn** [comment]
@SrNetoChan , this regression was most likely caused by your series of commits (commit 8b052fd and 6a1b572) during the 3.8.2 dev cycle, please review.

**Comment by SrNetoChan** [comment]
@nirvn those two commits were later changed by @nyalldawson https://github.com/qgis/QGIS/blame/master/src/gui/qgsmaptip.cpp#L149

```
  int cursorOffset = 0;
  // attempt to shift the tip away from the cursor.
  if ( QgsApplication::instance() )
  {
    // The following calculations are taken
    // from QgsApplication::getThemeCursor, and are used to calculate the correct cursor size
    // for both hi-dpi and non-hi-dpi screens.
    double scale = Qgis::UI_SCALE_FACTOR * QgsApplication::instance()->fontMetrics().height() / 32.0;
    cursorOffset = static_cast< int >( std::ceil( scale * 32 ) );
  }

  mWidget->move( pixelPosition.x() + cursorOffset, pixelPosition.y() );
```

Unfortunately, I don't fully understand the changes but, If this is really the cause of the problem, my guess is that the cursorOffset calculation and refresh is triggered as soon as the cursor moves, other than that, the mWidget->move part is almost the same as before, just adds a bit of spacing between the cursor and the tooltip.

**Comment by nyalldawson** [comment]
I suspect the issue is caused by moving the tip away from the cursor - previously because it was directly under it you had a way to move the cursor onto the tip without touching the map (and causing the tip to be cleared). Now, because there's a gap, it's impossible to move the cursor onto the tip's content without touching the canvas first. Basically the two fixes are mutually exclusive -- we'd need to roll back the shifting approach, OR look into whether we could use a timer to delay clearing tips for a short timeout.

**Comment by SrNetoChan** [comment]
Uhmm, that makes sense. The reason why I suggested the moving behavior was because you could hardly see the tip beginning with the cursor on top of it. The problem appeared with small tips like ids. maybe we can remove the moving behavior, and replace it by bigger padding or margin around the tip text.
If no solution is found, then it's better to remove the tip moving.

**Comment by northwestview** [comment]
If the cursor covering part of the information in the Map Tip is an issue,
this can be resolved by either the user moving the cursor slightly into the
map tip dialog. Alternatively,   inserting a blank space at the top of the
map tip will shift the information down and it will not be covered by the
cursor. I would prefer a reversion to the behaviour in 3.8.1. Other
solutions will create other issues I suspect.

**Comment by nirvn** [comment]
@SrNetoChan , @nyalldawson , would be good to try and address that prior to the next point release. If we want more time to think of a proper solution, I'd say let's remove the tip displacement for now.

**Comment by SrNetoChan** [comment]
@nirvn you are right. I will revert it.

**Comment by SrNetoChan** [comment]
This as been fixed by #31515 Can someone close it, please?
